chromedp_test.go

```diff
@@ -333,12 +333,20 @@ func TestConcurrentCancel(t *testing.T) {
 		ExecPath("/do-not-run-chrome"))
 	defer cancel()
 
+	var wg sync.WaitGroup
 	// 50 is enough for 'go test -race' to easily spot issues.
 	for i := 0; i < 50; i++ {
+		wg.Add(1)
 		ctx, cancel := NewContext(allocCtx)
-		go cancel()
-		go Run(ctx)
+		go func(wg *sync.WaitGroup) {
+			_ = Run(ctx)
+			wg.Done()
+		}(&wg)
+		go func() {
+			cancel()
+		}()
 	}
+	wg.Wait()
```

Comment:
How about changing it like this?

```go
	var wg sync.WaitGroup
	// 50 is enough for 'go test -race' to easily spot issues.
	for i := 0; i < 50; i++ {
		wg.Add(2)
		ctx, cancel := NewContext(allocCtx)
		go func() {
			cancel()
			wg.Done()
		}()
		go func() {
			Run(ctx)
			wg.Done()
		}()
	}
	wg.Wait()
```

1. `wg` can be used in the goroutine directly.
2. I think it should wait for `cancel()` to finish too.
3. I don't know why `cancel` is called before `Run`, but I think it's better to keep it (unless there is a reason to swap the order).

Reply:
Agree. Changed.

Reply:
> 3. I don't know why cancel is called before Run, but I think it's better to keep it (unless there is a reason to swap the order).

This one is not addressed.

Reply:
Changed.

Technically, goroutines don't run in the order they appeared in the code. The order is random.